### PR TITLE
Add CoreDHCP (and Tftp) Container(s) to Quickstart

### PR DESCRIPTION
**Pull request by synackd**
Add containers and configuration for CoreDHCP (built with the [coresmd](https://github.com/OpenCHAMI/coresmd) plugin) as well as a TFTP container to serve the relevant files. This is meant to replace dnsmasq, though it's files/config are kept around here for posterity.

Also, update the `generate-configs.sh` script to populate CoreDHCP configuration file, as well as update the README with relevant instructions.

### CODE REVIEW

File: quickstart/generate-configs.sh
Comment (davidallendj) [inline quickstart/generate-configs.sh:1]: When running `./generate-configs.sh -h` in the README, I'm getting errors saying to delete files and run again after the initial run:
```bash
Usage: [options] ./generate-configs.sh system-name [system-domain]
Example: ./generate-configs.sh foobar openchami.cluster
Example: ./generate-configs.sh foobar
Example: ./generate-configs.sh \
  -c 30s \
  -d 1.1.1.1,8.8.8.8 \
  -g 172.16.0.254 \
  -i 172.16.0.1 \
  -j 172.16.0.252 \
  -l 1h \
  -m 5m \
  -m 255.255.255.0 \
  -s 172.16.0.253 \
  -u http://172.16.0.253:8081 \
  foobar \
  openchami.cluster

Generate configuration for OpenCHAMI quickstart.

OPTIONS:
 -c  Duration of DHCP cache validity. Defaults to 30s.
 -d  Comma-separated list of DNS servers to use for DHCP. Defaults
     to 8.8.8.8.
 -f  Force overwriting config files.
 -g  DHCP gateway IP. Defaults of value of LOCAL_IP in generated
     .env file.
 -h  Print this usage message to stdout.
 -k  DHCP long lease time. 1 hour by default.
 -l  DHCP short lease time. 5 minutes by default.
 -m  DHCP netmask. Defaults to 255.255.255.0.
 -s  DHCP server IP. Defaults to value of LOCAL_IP in generated
     .env file.
 -u  Base URL for fetching boot scripts. Defaults to
     http://${LOCAL_IP}:8081.

ARGUMENTS:
 system-name   Subdomain of system to use in certificate and config
               generation. E.g. <system-name>.openchami.cluster
 system-domain (OPTIONAL) Domain of system to use in certificate and
               config generation. Defaults to openchami.cluster
A config file (.env) exists. Delete to generate a new one or -f to overwrite.
An OPAAL config (configs/opaal.yaml) exists. Delete to generate a new one or -f to overwrite.
A CoreDHCP config (configs/coredhcp.yaml) exists. Delete to generate a new one or -f to overwrite.
```
Can we run the help without have these pop up?

Comment (synackd) [reply quickstart/generate-configs.sh:1]: Ah, forgot to exit after printing the usage. Fixed in ab16328.

**Comment by davidallendj** [comment]
Quick note: 
When running the current deployment-recipes, I get the following error related to dnsmasq:
```bash
Pulling dnsmasq (ghcr.io/openchami/dnsmasq:v0.1.10)...
v0.1.10: Pulling from openchami/dnsmasq
ERROR: no matching manifest for linux/arm64/v8 in the manifest list entries
```
So I'm wondering if merging this PR will also inadvertently add support to running the recipes on Linux ARM64 as well.

**Comment by davidallendj** [comment]
> Quick note: When running the current deployment-recipes, I get the following error related to dnsmasq:
> 
> ```shell
> Pulling dnsmasq (ghcr.io/openchami/dnsmasq:v0.1.10)...
> v0.1.10: Pulling from openchami/dnsmasq
> ERROR: no matching manifest for linux/arm64/v8 in the manifest list entries
> ```
> 
> So I'm wondering if merging this PR will also inadvertently add support for running the recipes on Linux ARM64 as well.

Just to answer my own question, I wasn't able to run the recipes on the raspberry PI due to the same error but with `coresmd` instead. I imagine when we get multi-arch support going, this will eventually be solved.

**Comment by synackd** [comment]
Yeah, it's only building amd64 containers in CI right now, so I can try to update that for multiarch. Should work after that.

File: quickstart/generate-configs.sh
Comment (alexlovelltroy) [inline quickstart/generate-configs.sh:1]: generate-configs.sh is now over 200 lines long and has become a "load-bearing bash script".  We should consider alternate approaches that are easier to isolate and test.

**Comment by synackd** [comment]
I've reduced the amount of things that the generate-configs script does and modified the quickstart README with the assumptions/defaults.

**Pull request merged**